Mathlib/MeasureTheory/Integral/Asymptotics.lean

```diff
@@ -38,3 +48,91 @@ theorem _root_.Asymptotics.IsBigO.integrable (hfm : AEStronglyMeasurable f μ)
+  all_goals exact hf.aestronglyMeasurable.stronglyMeasurableAtFilter
+
+/-- If `f` is locally integrable on `(∞, a]`, and `f =O[atBot] g`, for some
+`g` integrable at `atBot`, then `f` is integrable on `(∞, a]`. -/
+theorem LocallyIntegrableOn.integrableOn_of_isBigO_atBot [IsMeasurablyGenerated (atBot (α := α))]
+    (hf : LocallyIntegrableOn f (Iic a) μ) (ho : f =O[atBot] g)
+    (hg : IntegrableAtFilter g atBot μ) : IntegrableOn f (Iic a) μ := by
+  refine integrableOn_Iic_iff_integrableAtFilter_atBot.mpr ⟨ho.integrableAtFilter ?_ hg, hf⟩
+  exact ⟨Iic a, Iic_mem_atBot a, hf.aestronglyMeasurable⟩
+
+/-- If `f` is locally integrable on `[a, ∞)`, and `f =O[atTop] g`, for some
+`g` integrable at `atTop`, then `f` is integrable on `[a, ∞)`. -/
+theorem LocallyIntegrableOn.integrableOn_of_isBigO_atTop [IsMeasurablyGenerated (atTop (α := α))]
+    (hf : LocallyIntegrableOn f (Ici a) μ) (ho : f =O[atTop] g)
+    (hg : IntegrableAtFilter g atTop μ) : IntegrableOn f (Ici a) μ := by
+  refine integrableOn_Ici_iff_integrableAtFilter_atTop.mpr ⟨ho.integrableAtFilter ?_ hg, hf⟩
+  exact ⟨Ici a, Ici_mem_atTop a, hf.aestronglyMeasurable⟩
+
+/-- If `f` is locally integrable, `f` has a top element, and `f =O[atBot] g`, for some
+`g` integrable at `atBot`, then `f` is integrable. -/
+theorem LocallyIntegrable.integrable_of_isBigO_atBot [IsMeasurablyGenerated (atBot (α := α))]
+    [OrderTop α] (hf : LocallyIntegrable f μ) (ho : f =O[atBot] g)
+    (hg : IntegrableAtFilter g atBot μ) : Integrable f μ := by
+  refine integrable_iff_integrableAtFilter_atBot.mpr ⟨ho.integrableAtFilter ?_ hg, hf⟩
+  exact hf.aestronglyMeasurable.stronglyMeasurableAtFilter
+
+/-- If `f` is locally integrable, `f` has a bottom element, and `f =O[atTop] g`, for some
+`g` integrable at `atTop`, then `f` is integrable. -/
+theorem LocallyIntegrable.integrable_of_isBigO_atTop [IsMeasurablyGenerated (atTop (α := α))]
+    [OrderBot α] (hf : LocallyIntegrable f μ) (ho : f =O[atTop] g)
+    (hg : IntegrableAtFilter g atTop μ) : Integrable f μ := by
+  refine integrable_iff_integrableAtFilter_atTop.mpr ⟨ho.integrableAtFilter ?_ hg, hf⟩
+  exact hf.aestronglyMeasurable.stronglyMeasurableAtFilter
+
+end LinearOrder
+
+section LinearOrderedAddCommGroup
+
+variable [LinearOrderedAddCommGroup α] [CompactIccSpace α]
+
+/-- If `f` is locally integrable, `‖f(-x)‖ = ‖f(x)‖`, and `f =O[atTop] g`, for some
+`g` integrable at `atTop`, then `f` is integrable. -/
+theorem LocallyIntegrable.integrable_of_isBigO_atTop_of_norm_isNegInvariant
+    [IsMeasurablyGenerated (atTop (α := α))] [MeasurableNeg α] [μ.IsNegInvariant]
+    (hf : LocallyIntegrable f μ) (hsymm : norm ∘ f =ᵐ[μ] norm ∘ f ∘ Neg.neg) (ho : f =O[atTop] g)
+    (hg : IntegrableAtFilter g atTop μ) : Integrable f μ := by
+  refine (isEmpty_or_nonempty α).casesOn (fun _ ↦ ?_) (fun _ ↦ ?_)
+  · exact integrableOn_univ.mp (by convert integrableOn_empty)
+  let a := -|Classical.arbitrary α|
```

Comment:
This is unnecessary. By hypothesis `α` is a `LinearOrderedAddCommGroup`; so it can't be empty, it has to have a zero (indeed mathlib will happily synthesise a `Nonempty α` instance in this context if you ask it to). Similarly you don't have to invoke `Classical.arbitrary` to get your hands on an element of `α`, you can just use 0.

Reply:
Nice catch, removed

Reply:
Can you also remove the `isEmpty_or_nonempty α` lines, since they are not needed?

Reply:
Of course, removed